### PR TITLE
Ftr/11408 checkbox treatment 1

### PR DESCRIPTION
**Pull request by jagu65**
**High level description:**
Create a common component for the checkbox treatment #1

**Technical details:**
New common components made utilizing code from Recipient Type and PrimaryCheckboxType.jsx

**JIRA Ticket:**
[DEV-11408](https://federal-spending-transparency.atlassian.net/browse/DEV-11408)

**Mockup:**
https://zpl.io/Bwm9WEp

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
- [ ] Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
- [ ] Verified cross-browser compatibility: Chrome, Safari, Firefox, Edge
- [ ] Verified mobile/tablet/desktop/monitor responsiveness
- [ ] Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [ ] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
- [ ] [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
- [ ] [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
- [ ] Design review complete `if applicable`
- [ ] [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/js/components/sharedComponents/checkbox/CheckboxOnePrimary.jsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
+/**
+ * CheckboxOnePrimary.jsx
```
Comment (anjenkin) [inline src/js/components/sharedComponents/checkbox/CheckboxOnePrimary.jsx:2]: Let's change the name from CheckboxOne to something more descriptive.  A year from now no one is going to know what treatment #1 means.  This treatment uses accordions so AccordionCheckbox could be a possible name.

Comment (jagu65) [reply src/js/components/sharedComponents/checkbox/CheckboxOnePrimary.jsx]: Renamed to Accordion Checkbox!

File: src/js/components/search/filters/recipient/RecipientTypeAccordion.jsx [hunk 2]
```diff
@@ -141,6 +142,7 @@ const RecipientTypeAccordion = ({
             <div className="checkbox-type-filter">
                 <ul className="checkbox-types">
                     {checkboxTypes}
+                    {console.log('selectedTypes: ', selectedTypes)}
```
Comment (brianpetway) [inline src/js/components/search/filters/recipient/RecipientTypeAccordion.jsx:145]: console.log left in

File: src/js/components/sharedComponents/checkbox/AccordionCheckbox.jsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/**
+ * CheckboxOnePrimary.jsx
```
Comment (brianpetway) [inline src/js/components/sharedComponents/checkbox/AccordionCheckbox.jsx:2]: need to rename to match the component name

File: src/js/components/sharedComponents/checkbox/AccordionCheckboxSecondary.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+/**
+ * CheckboxOnePrimary.jsx
```
Comment (brianpetway) [inline src/js/components/sharedComponents/checkbox/AccordionCheckboxSecondary.jsx:2]: need to rename to match the component name

File: src/js/components/sharedComponents/checkbox/AccordionCheckboxPrimary.jsx [hunk 5]
```diff
@@ -0,0 +1,160 @@
+/**
+ * CheckboxOnePrimary.jsx
```
Comment (brianpetway) [inline src/js/components/sharedComponents/checkbox/AccordionCheckboxPrimary.jsx:2]: need to rename to match the component name

File: src/js/components/search/filters/recipient/RecipientTypeAccordion.jsx (hunk 2 above)
Comment (jagu65) [reply src/js/components/search/filters/recipient/RecipientTypeAccordion.jsx]: removed

File: src/js/components/sharedComponents/checkbox/AccordionCheckbox.jsx (hunk 3 above)
Comment (jagu65) [reply src/js/components/sharedComponents/checkbox/AccordionCheckbox.jsx]: renamed

File: src/js/components/sharedComponents/checkbox/AccordionCheckboxSecondary.jsx (hunk 4 above)
Comment (jagu65) [reply src/js/components/sharedComponents/checkbox/AccordionCheckboxSecondary.jsx]: renamed

File: src/js/components/sharedComponents/checkbox/AccordionCheckboxPrimary.jsx (hunk 5 above)
Comment (jagu65) [reply src/js/components/sharedComponents/checkbox/AccordionCheckboxPrimary.jsx]: renamed

File: src/_scss/pages/search/searchPage.scss [hunk 6]
```diff
@@ -15,6 +15,8 @@
     @import "resultsView/resultsView";
     @import "resultsView/dtuiOverrides";
 
+    @import "components/accordionCheckbox";
```
Comment (brianpetway) [inline src/_scss/pages/search/searchPage.scss:18]: is this import necessary?

File: src/_scss/components/_accordionCheckbox.scss [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+      svg {
+        color: $blue-50;
+        max-width: rem(10);
+        max-height: rem(10);
+      }
+
+
+      input {
+        margin-left: rem(8);
```
Comment (brianpetway) [inline src/_scss/components/_accordionCheckbox.scss:19]: should use the var for global-margin for these

File: src/_scss/pages/search/searchPage.scss (hunk 6 above)
Comment (jagu65) [reply src/_scss/pages/search/searchPage.scss:18]: Yes, temp-page uses this file for its styling

File: src/_scss/components/_accordionCheckbox.scss (hunk 7 above)
Comment (jagu65) [reply src/_scss/components/_accordionCheckbox.scss]: global margin variable used to replace generic rems

**Review by brianpetway — APPROVED** [review]

**Pull request merged**